discovery-infra/test_infra/tools/concurrently.py

```diff
@@ -1,8 +1,9 @@
 import logging
 from concurrent.futures import ThreadPoolExecutor
+from typing import Callable, Dict, List, Tuple, Union
 
 
-def _safe_run(job, job_id, done_handler):
+def _safe_run(job, job_id: int, done_handler: Callable[[int], None]):
```

Comment:
Use `Optional`

Reply:
Where? `Callable[[int], None]` annotate the function type so done handler signature must be `handler(x: int) -> None`